### PR TITLE
feat: Deep connector Zendesk Support (Metadata)

### PR DESCRIPTION
**Pull request by Cobalt0s**
Closes https://github.com/amp-labs/connectors/issues/616
Merge after https://github.com/amp-labs/connectors/pull/630

# OpenAPI

Static file is served for ListObjectMetadata. It is created by extracting metadata using OpenAPI file downloaded from [Zendesk website](https://developer.zendesk.com/api-reference/ticketing/introduction/#download-openapi-file). 

APIs included:
* :heavy_check_mark: Ticketing
* :heavy_check_mark: Help Desk
* :x: Voice (phone-numbers endpoint maybe the only one relevarnt, others are not List Read) 
image

# Tools

New tool was introduced in this PR as it is the first Connector to derive metadata from OpenAPI.
This tool can be reused for future connectors that rely on OpenAPI. It is located under:
```
tools/fileconv/api3 => OpenapiFileManager
```
This file manager loads data under `zendesksupport/openapi/file.go`. Later used to generate `zendesksupport/metadata/schema.json` by `scripts/scrapper/zendesksupport/main.go`.

# ListObjectMetadata
Running `zendesksupport/metadata/main.go` will read **brands** and compare if fields are part of expected ListObjectMetadata response.
image

### CODE REVIEW

**Comment by Cobalt0s** [comment]
* **#631** <a href="https://app.graphite.dev/github/pr/amp-labs/connectors/631?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `cobalt0s/zendesk-support`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Cobalt0s and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by Cobalt0s** [comment]
@eberle1080  

Double checked Read/Write/Metadata scripts. All of them are passing.

image

image

image

**Review by laurenzlong — CHANGES_REQUESTED** [review]
@Cobalt0s I'm a bit confused the need for the aliases package, can you add comments (to `zendesksupport/metadata.go` to explain why we need to know about aliases in order to retrieve Zendesk object metadata? And if it is really needed by the Zendesk connector, can you move everything in handy/aliases.go to the zendesk connector? I am quite doubtful that another connector will need the aliases function.

File: zendesksupport/openapi/file.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"github.com/amp-labs/connectors/tools/fileconv/api3"
+)
+
+var (
+	// Static file containing openapi spec.
+	//
+	//go:embed support-api.yaml
+	apiFile []byte
+
+	FileManager = api3.NewOpenapiFileManager(apiFile) // nolint:gochecknoglobals
```
Comment (Cobalt0s) [inline zendesksupport/openapi/file.go:15]: `support-api.yaml` is loaded into FileManager.

The script uses this FileManager along with the OpenAPI tool to produce `schemas.json`.

File: scripts/scrapper/zendesksupport/metadata/main.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+func main() {
+	explorer, err := openapi.FileManager.GetExplorer()
+	must(err)
+
+	aliases := api3.NewAliases(map[string]string{
+		"search":               "results",
+		"ticket_audits":        "audits",
+		"satisfaction_reasons": "reasons",
+		// problems -> additionalProperties
```
Comment (Cobalt0s) [inline scripts/scrapper/zendesksupport/metadata/main.go:20]: Some object names are stored under different alternative naming within openapi schema.
For Zendesk there are 3 exceptions.

Problems has no static fields, it is a free form map.

**Review by Cobalt0s — COMMENTED** [review]
@eberle1080 

This PR was split into smaller chunks for easier review.
It also makes it explicit on how Zendesk Support and Pipeliner are related. Both use OpenAPI tool: https://github.com/amp-labs/connectors/pull/718

**Pull request closed without merging**